extensions/mapstore/src/test/java/com/hazelcast/mapstore/GenericMapStoreTest.java

```diff
@@ -122,22 +133,23 @@ public void whenMapStoreDestroyOnMaster_thenDropMapping() throws Exception {
         createTable(mapName);
 
         mapStore = createMapStore();
-        awaitMappingCreated();
+        assertMappingCreated();
 
         mapStore.destroy();
-        awaitMappingDestroyed();
+        assertMappingDestroyed();
     }
 
     @Test
     public void whenMapStoreDestroyOnNonMaster_thenDropMapping() throws Exception {
         createTable(mapName);
 
         mapStore = createMapStore();
-        awaitMappingCreated();
+        assertMappingCreated();
 
         GenericMapStore<Object> mapStoreNotMaster = createMapStore(instances()[1]);
+        mapStoreNotMaster.awaitInitFinished();
         mapStoreNotMaster.destroy();
```

Comment:
`createMapStore` and `GenericMapStore#destroy` do `awaitInitFinished` already.

Reply:
Changed.

Also, I don't have strong opinion on the defensive checks, but I see your point, so I'm removing this defensive mapping removal